app/uploaders/evss_claim_document_uploader.rb

```diff
@@ -6,15 +6,45 @@ class EVSSClaimDocumentUploader < CarrierWave::Uploader::Base
   include SetAWSConfig
   include ValidateEVSSFileSize
 
+  class << self
+    def tiff?(mimemagic_object:, carrier_wave_sanitized_file:)
+      mimemagic_object&.type == 'image/tiff' || carrier_wave_sanitized_file&.content_type == 'image/tiff'
+    end
+
+    def incorrect_extension?(extension:, mimemagic_object:)
+      extension = extension.to_s.downcase
+      true_extensions = extensions_from_mimemagic_object(mimemagic_object).map(&:downcase)
+      true_extensions.present? && !extension.in?(true_extensions)
+    end
+
+    def extensions_from_mimemagic_object(mimemagic_object)
+      mimemagic_object&.extensions || []
+    end
+
+    def inspect_binary(carrier_wave_sanitized_file)
+      file_obj = carrier_wave_sanitized_file&.to_file
+      file_obj && MimeMagic.by_magic(file_obj)
+    ensure
+      file_obj&.close
+    end
+  end
+
   def size_range
     1.byte...150.megabytes
   end
 
-  version :converted, if: :tiff? do
-    process(convert: :jpg)
+  version :converted, if: :tiff_or_incorrect_extension? do
+    process(convert: :jpg, if: :tiff?)
 
-    def full_filename(file)
-      "converted_#{file}.jpg"
+    def full_filename(original_name_for_file)
+      name = "converted_#{original_name_for_file}"
+      extension = CarrierWave::SanitizedFile.new(nil).send(:split_extension, original_name_for_file)[1]
```

Comment:
not in love with calling a libraries private method here, but I want to use same file-ext-extraction code as CarrierWave